### PR TITLE
Bugfix lwdev 7232 depth chart rendering improvements

### PR DESCRIPTION
**Pull request by GeorgePuisha**
https://lykkex.atlassian.net/browse/LWDEV-7232
and many others

### CODE REVIEW

File: src/components/DepthChart/Chart/Chart.tsx [hunk 1]
```diff
@@ -213,14 +223,15 @@ class Chart extends React.Component<ChartProps> {
     this.graphics = [];
     this.width = this.props.width;
     this.height = this.props.height;
-    this.midXAsks = this.width / 2 + Math.round(chart.strokeWidth / 2);
-    this.midXBids = this.width / 2 - Math.round(chart.strokeWidth / 2);
+    this.midXAsks = this.props.width / 2 + Math.round(chart.strokeWidth / 2);
+    this.midXBids = this.props.width / 2 - Math.round(chart.strokeWidth / 2);
     this.midY = this.height;
-    this.asks = this.props.asks.reverse();
+    this.asks = reverse(this.props.asks);
```
Comment (sergeyschegolev) [inline src/components/DepthChart/Chart/Chart.tsx]: is it necessary to define these fields? you can get it directly from the props in your methods. If you need to reverse asks you can do it in the index.tsx

File: src/components/DepthChart/Chart/Chart.tsx [hunk 2]
```diff
@@ -4,11 +4,22 @@ import {Line} from 'react-konva';
...
+  bids: Order[];
+  mid: number;
+  width: number;
+  height: number;
+  quoteAccuracy: number;
+  baseAccuracy: number;
+  priceAccuracy: number;
+}
+
 class Chart extends React.Component<ChartProps> {
```
Comment (sergeyschegolev) [inline src/components/DepthChart/Chart/Chart.tsx]: you should use stateless component as far as you don't need a state

File: src/components/DepthChart/Chart/ChartWrapper.tsx [hunk 3]
```diff
@@ -1,29 +1,26 @@
...
+export interface ChartWrapperProps {
+  setWidth: (width: number) => void;
+  setHeight: (height: number) => void;
+  selectedInstrument?: any;
+}
 
-class ChartWrapper extends React.Component<ChartProps> {
-  @observable width: number = -1;
-  @observable height: number = -1;
+class ChartWrapper extends React.Component<ChartWrapperProps> {
```
Comment (sergeyschegolev) [inline src/components/DepthChart/Chart/ChartWrapper.tsx]: The same here: you can use stateless component

File: src/components/DepthChart/Chart/Mesh.tsx [hunk 4]
```diff
@@ -77,11 +94,13 @@ class Mesh extends React.Component<ChartProps> {
...
-    const maximum = this.calculateMaxDepth() / chart.scaleFactor;
 
-    const step = maximum / chart.mesh.horizontalLinesAmount;
-    for (let i = 0; i < chart.mesh.horizontalLinesAmount; i++) {
-      labels.push(formattedNumber(step * (i + 1) - step / 2, 2));
+    if (this.asks.length > 0 || this.bids.length > 0) {
+      const maximum = this.calculateMaxDepth() / chart.scaleFactor;
+      const step = maximum / chart.mesh.horizontalLinesAmount;
+      for (let i = 0; i < chart.mesh.horizontalLinesAmount; i++) {
+        labels.push(formattedNumber(step * (i + 1) - step / 2, 2));
```
Comment (sergeyschegolev) [inline src/components/DepthChart/Chart/Mesh.tsx]: As I remember accuracy should be this.props.baseAccuracy. BTW there is a right value in the dev branch

File: src/components/DepthChart/Chart/chartHelpers.ts [hunk 5]
```diff
@@ -1,9 +1,5 @@
...
-  measureText = (text: string, size: number, font: string) => {
-    const ctx = document.createElement('canvas').getContext('2d');
-    ctx!.font = `${size}px ${font}`;
-    return Math.ceil(ctx!.measureText(text).width);
-  };
-}
-
-export default ChartHelpers;
+export const measureText = (text: string, size: number, font: string) => {
+  const ctx = document.createElement('canvas').getContext('2d');
```
Comment (sergeyschegolev) [inline src/components/DepthChart/Chart/chartHelpers.ts]: Is it possible to create a canvas element once outside this function?

File: src/stores/depthChartStore.ts [hunk 6]
```diff
@@ -39,17 +43,19 @@ class DepthChartStore extends BaseStore {
   }
 
   reduceBidsArray = (bids: Order[]) => {
-    const mid = this.mid();
-    const lowerBound = mid - mid * this.multiplers[this.spanMultiplierIdx];
+    const maximum = Math.max(...bids.map(b => b.price));
+    const lowerBound =
+      maximum - maximum * this.multiplers[this.spanMultiplierIdx];
     const filteredBids = bids.filter(bid => {
```
Comment (sergeyschegolev) [inline src/stores/depthChartStore.ts]: return is not necessary. just  bids.filter(bid => bid.price > lowerBound)

File: src/components/DepthChart/Chart/Chart.tsx [hunk 7]
```diff
@@ -183,45 +193,37 @@ class Chart extends React.Component<ChartProps> {
...
-        priceAccuracy={this.props.priceAccuracy}
+        orders={this.props.bids}
+        points={this.pointsBids}
+        borders={[0, this.props.height, this.midXBids, 0]}
       />
     );
   };
 
-  calculateCoef() {
+  calculatecoefficient() {
```
Comment (optimusway) [inline src/components/DepthChart/Chart/Chart.tsx]: `calculateCoefficient`

File: src/components/DepthChart/Chart/index.tsx [hunk 8]
```diff
@@ -1,20 +1,85 @@
...
+    return {
+      setWidth,
+      setHeight,
+      selectedInstrument
+    };
+  },
+  ChartWrapper
+);
 
+const ConnectedMesh = connect(
```
Comment (optimusway) [inline src/components/DepthChart/Chart/index.tsx]: 👍

**Pull request merged**